client/app/types/course/courses.ts

```diff
@@ -71,3 +74,35 @@ export interface NewCourseFormData {
   title: string;
   description: string;
 }
+
+export interface SidebarItemData {
```

Comment:
I didn't pass the `weight` to the front-end because if the only use of `weight` is to determine ordering, the array of `SidebarItemData` is already sorted according to the weights from the back-end. In this case, there's no need for the front-end to know that `weight` exists.